PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = 'iap.account'
-    _inherit = 'ir.model'
-    _inherit = 'mail.message'
-    _inherit = 'mail.notification'

Total Changes: 23

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/sms/models/iap_account.py
+++ b/addons/sms/models/iap_account.py
@@ -4,7 +4,7 @@ from odoo import fields, models, _
 
 
 class IapAccount(models.Model):
-    _inherit = 'iap.account'
+    _inherit = ['iap.account']
 
     sender_name = fields.Char(help="This is the name that will be displayed as the sender of the SMS.", readonly=True)
 

--- a/addons/sms/models/ir_actions_server.py
+++ b/addons/sms/models/ir_actions_server.py
@@ -5,9 +5,8 @@ from odoo import _, api, fields, models
 from odoo.exceptions import ValidationError
 
 
-class ServerActions(models.Model):
+class IrActionsServer(models.Model):
     """ Add SMS option in server actions. """
-    _name = 'ir.actions.server'
     _inherit = ['ir.actions.server']
 
     state = fields.Selection(selection_add=[
@@ -33,7 +32,7 @@ class ServerActions(models.Model):
             mail_models = self.env['ir.model'].search([('is_mail_thread', '=', True), ('transient', '=', False)])
             for action in mail_thread_based:
                 action.available_model_ids = mail_models.ids
-        super(ServerActions, self - mail_thread_based)._compute_available_model_ids()
+        super(IrActionsServer, self - mail_thread_based)._compute_available_model_ids()
 
     @api.depends('model_id', 'state')
     def _compute_sms_template_id(self):

--- a/addons/sms/models/ir_model.py
+++ b/addons/sms/models/ir_model.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class IrModel(models.Model):
-    _inherit = 'ir.model'
+    _inherit = ['ir.model']
 
     is_mail_thread_sms = fields.Boolean(
         string="Mail Thread SMS", default=False,

--- a/addons/sms/models/mail_message.py
+++ b/addons/sms/models/mail_message.py
@@ -7,7 +7,7 @@ class MailMessage(models.Model):
     """ Override MailMessage class in order to add a new type: SMS messages.
     Those messages comes with their own notification method, using SMS
     gateway. """
-    _inherit = 'mail.message'
+    _inherit = ['mail.message']
 
     message_type = fields.Selection(
         selection_add=[('sms', 'SMS')],

--- a/addons/sms/models/mail_notification.py
+++ b/addons/sms/models/mail_notification.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class MailNotification(models.Model):
-    _inherit = 'mail.notification'
+    _inherit = ['mail.notification']
 
     notification_type = fields.Selection(selection_add=[
         ('sms', 'SMS')

--- a/addons/sms/models/mail_thread.py
+++ b/addons/sms/models/mail_thread.py
@@ -10,7 +10,7 @@ _logger = logging.getLogger(__name__)
 
 
 class MailThread(models.AbstractModel):
-    _inherit = 'mail.thread'
+    _inherit = ['mail.thread']
 
     message_has_sms_error = fields.Boolean(
         'SMS Delivery error', compute='_compute_message_has_sms_error', search='_search_message_has_sms_error',

--- a/addons/sms/models/res_partner.py
+++ b/addons/sms/models/res_partner.py
@@ -5,5 +5,4 @@ from odoo import models
 
 
 class ResPartner(models.Model):
-    _name = 'res.partner'
     _inherit = ['mail.thread.phone', 'res.partner']

--- a/addons/sms/models/sms_sms.py
+++ b/addons/sms/models/sms_sms.py
@@ -14,7 +14,6 @@ _logger = logging.getLogger(__name__)
 
 
 class SmsSms(models.Model):
-    _name = 'sms.sms'
     _description = 'Outgoing SMS'
     _rec_name = 'number'
     _order = 'id DESC'

--- a/addons/sms/models/sms_template.py
+++ b/addons/sms/models/sms_template.py
@@ -4,9 +4,8 @@
 from odoo import api, fields, models, _
 
 
-class SMSTemplate(models.Model):
+class SmsTemplate(models.Model):
     "Templates for sending SMS"
-    _name = "sms.template"
     _inherit = ['mail.render.mixin', 'template.reset.mixin']
     _description = 'SMS Templates'
 
@@ -47,7 +46,7 @@ class SMSTemplate(models.Model):
 
     def unlink(self):
         self.sudo().mapped('sidebar_action_id').unlink()
-        return super(SMSTemplate, self).unlink()
+        return super().unlink()
 
     def action_create_sidebar_action(self):
         ActWindow = self.env['ir.actions.act_window']

--- a/addons/sms/models/sms_tracker.py
+++ b/addons/sms/models/sms_tracker.py
@@ -16,7 +16,6 @@ class SmsTracker(models.Model):
     Note: Only admins/system user should need to access (a fortiori modify) these technical
       records so no "sudo" is used nor should be required here.
     """
-    _name = 'sms.tracker'
     _description = "Link SMS to mailing/sms tracking models"
 
     SMS_STATE_TO_NOTIFICATION_STATUS = {
